hmcts/pre-api
# S28 3098: End Live Event (AMS Implementation)

### Change description
- Add `PUT /media-service/live-event/end/{captureSessionId}` endpoint
- Add AMS implementation for stopping live event




> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- General step-up steps
  - Create a capture session using PUT /capture-sessions/{id} endpoint
  - Start a live event (using PUT /media-service/live-events/start/{captureSessionId}
  - Create a streaming endpoint + locator (either using endpoint if merged in at the time of testing or manually via the Azure portal). 
- Test: Create + run a CVP recording for a few mins and then use the /media-service/live-event/stop/{captureSessionId} endpoint to end the recording + create the asset. See no error response, in azure portal see related streaming endpoint + locator and live event are all stopped and deleted. Using the /recordings?captureSessionId={captureSessionId} endpoint see a newly created (v1) recording. See capture session has been updated with finished at/by values + status is RECORDING_AVAILABLE. In azure portal, see newly created asset- stored in final storage account
- Test: Do NOT create a CVP recording. Call stop endpoint. See no error, see in azure portal that related streaming endpoint, locator and live event are stopped and deleted. Using the /recordings?captureSessionId={captureSessionId} get no values in the response. See capture session has been updated with finished at/by values + status is NO_RECORDING. 
- Test: Attempt to stop a live event for a capture session that does not exist- see 404
- Test: Attempt to stop a live event for a capture session that has been deleted- see 404
- Test: Attempt to stop a live event that has already been finished (finished at is not null)- see 200
- Test: Attempt to stop a live event that has not been started (started at is null)- see 400
- Test: Attempt to stop a live event that has status that is not RECORDING or STANDBY- see 400
- Test: Start live event and then manually stop it in azure portal, now call stop endpoint. See error and capture session has been updated with finished at/by and status = FAILURE.
-  Any other tests (?)
